docs/xla_flags.md

```diff
@@ -0,0 +1,90 @@
+# XLA & Jax
+
+<!--* freshness: { reviewed: '2024-08-18' } *-->
+
+## Introduction
+This guide explains basics about XLA and how XLA related to Jax.
+For in-depth details please refer to [XLA documentation](https://openxla.org/xla)
+
+## XLA: The Powerhouse Behind JAX
+XLA (Accelerated Linear Algebra) is a domain-specific compiler for linear algebra that plays a pivotal role in JAX's performance and flexibility. It enables JAX to generate optimized code for various hardware backends (CPUs, GPUs, TPUs) by transforming and compiling your Python/NumPy-like code into efficient machine instructions.
+
+JAX uses XLA's JIT compilation capabilities to transform your Python functions into optimized XLA computations at runtime.
+
+## Configuring XLA in JAX:
+You can influence XLA's behavior in JAX by setting XLA_FLAGS environment variables before running your Python script or colab notebook.
+
+For the colab notebooks:
+
+Provide flags using `os.environ['XLA_FLAGS']`:
+
+
+```python
+import os
+
+# Set multiple flags separated by spaces
+os.environ['XLA_FLAGS'] = '--flag1=value1 --flag2=value2'
+```
+
+For the python scripts:
+
+Specify `XLA_FLAGS` as a part of cli command:
+
+```bash
+XLA_FLAGS='--flag1=value1 --flag2=value2'  python3 source.py
+```
+
+**Important Notes:**
+
+* Set XLA_FLAGS before importing JAX or other relevant libraries.
+* Refer to XLA and JAX documentation for a complete list of available flags.
+* Experiment with different flags to optimize performance for your specific use case.
+
+
+**For further information:**
+* Complete and up to date documentation about XLA can be found in the official [XLA documentation](https://openxla.org/xla).
+
+* For backends supported by open-source version of XLA (CPU, GPU), XLA flags are defined with their default values in [xla/debug_options_flags.cc](https://github.com/openxla/xla/blob/main/xla/debug_options_flags.cc), and complete list of flags could be found [here](https://github.com/openxla/xla/blob/main/xla/xla.proto).
+* However, [OpenXLA](https://github.com/openxla/xla) doesn't have configuration for the proprietary backends like TPU, so getting complete list of flags could be challenging.
+
+* Below, we describe a subset of backend-specific XLA flags that can be used to optimize the performance of your JAX program. Please note that this list is not exhaustive. Also, given the experimental nature of some flags and possible differences in backend versions, please check the documentation for your specific backend version and contact support if some flags do not work as expected.
+### Common XLA flags
+| Flag | Type | Notes |
+| ---- | ---- | ----- |
+| xla_dump_to | String (filepath) | The folder where pre-optimization HLO files and other artifacts will be placed (see [XLA Tools](https://openxla.org/xla/tools)). |
```

Comment:
Consider using backticks around flag names in order to render them in fixed-width, e.g. ``` `xla_dump_to` ```